backend/dataall/modules/redshift_datasets_shares/services/redshift_table_share_processor.py

```diff
@@ -18,11 +33,477 @@ def __init__(self, session, share_data, shareable_items, reapply=False):
+        """
+        1) (in source namespace) Check the datashare exists
+        2) (in source namespace) Check that schema is added to datashare
+        3) (in source namespace) Check the access is granted to the consumer cluster to the datashare
+        4) (in target namespace) Check that local db exists
+        5) (in target namespace) Check that the redshift role has access to the local db
+        6) (in target namespace) Check that external schema exists
+        7) (in target namespace) Check that the redshift role has access to the extenal schema
+        For each table:
+            8) (in source namespace) Check that table is added to datashare
+            9) (in target namespace) Check that the redshift role has select access to the requested table in the local db.
+            10) (in target namespace) Check that the redshift role has select access to the requested table in the external schema.
+        """
+
+        log.info('##### Verifying Redshift tables #######')
+        if not self.tables:
+            log.info('No tables to verify. Skipping...')
+        else:
+            tbl_level_errors = []
+            ds_level_errors = []
+            redshift_client_in_source = redshift_share_data_client(
+                account_id=self.share_data.source_environment.AwsAccountId,
+                region=self.share_data.source_environment.region,
+                connection=self.source_connection,
+            )
+            redshift_client_in_target = redshift_share_data_client(
+                account_id=self.share_data.target_environment.AwsAccountId,
+                region=self.share_data.target_environment.region,
+                connection=self.target_connection,
+            )
+            local_db = self._build_local_db_name()
+            external_schema = self._build_external_schema_name()
+            try:
+                # 1) (in source namespace) Check that datashare exists
+                if not redshift_client_in_source.check_datashare_exists(self.datashare_name):
+                    ds_level_errors.append(ShareErrorFormatter.dne_error_msg('Redshift datashare', self.datashare_name))
+                # 2) (in source namespace) Check that schema is added to datashare
+                if not redshift_client_in_source.check_schema_in_datashare(
+                    datashare=self.datashare_name, schema=self.dataset.schema
+                ):
+                    ds_level_errors.append(
+                        ShareErrorFormatter.dne_error_msg(
+                            'Redshift schema added to datashare',
+                            f'datashare_name={self.datashare_name}, schema={self.dataset.schema}',
+                        )
+                    )
+                # 3) (in source namespace) Check the access is granted to the consumer cluster to the datashare
+                if not redshift_client_in_source.check_consumer_permissions_to_datashare(datashare=self.datashare_name):
```

Comment:
should this be called from consumer (i.e. target) to see if datashare is accessible from consuemr side? Otherwise is this not similar to `check_datashare_exists` ?

Reply:
Yes, yes yes. Thank you!